### DISCUSSION TITLE
Why is draw rectangle leaving a gray border on my PDF?

### DISCUSSION THREAD

**Discussion by Saya47** [ImageMagick/ImageMagick#4332] [Help] [answered]
```
convert -density 300 doc.pdf -stroke white -strokewidth 0 -fill white -draw "rectangle 0,0 2467,80" -draw "rectangle 1034,3452 1417,3504" -quality 100 -density 300 -strip out.pdf
```
Here's the second rectangle:
![image](https://user-images.githubusercontent.com/35542834/136551589-622c6c43-481e-40ef-9849-7a9731f6839d.png)
And it's visible on Adobe Acrobat Reader, Chrome etc.

The issue does not happen if the input is a PNG file instead of a PDF.
How can I get rid of it?

**Comment by snibgo** [comment]
What version of IM, on what platform?

What version of Ghostscript?

Please post links to your input and output PDFs.

**Reply by Saya47** [reply]
Hello! Thanks for replying!
IM Version: ImageMagick 7.0.10-27 Q16 x86_64 2021-02-07
Platform: Windows, Cygwin.
GS version: GPL Ghostscript 9.53.3 (2020-10-01)
Input PDF: [Input.pdf](https://github.com/ImageMagick/ImageMagick/files/7319959/Input.pdf)
Output PDF: [Output.pdf](https://github.com/ImageMagick/ImageMagick/files/7319961/Output.pdf)

**Comment by snibgo** [comment, marked as answer]
Your input.pdf is mostly transparent. You are drawing white rectangles on this. Adobe Acrobat Reader by default shows transparency as white, and shows a gray border between "white that means white" and "white that means transparent".

You might "-background White -layers Flatten" immediately after reading the input PDF, so you have no transparency.

EDIT to add: In Adobe Acrobat Reader, if you don't want it to show a gray border between white and transparent, in Edit| Preferences | Rendering, untick the box for "smooth images".

**Reply by Saya47** [reply]
Thanks very much for the answer!
